Closes #1350: Implement more flexible update hook numbering scheme.

joeparsons opened the pull request:
## Description
- Adds update hook numbering convention guidelines to RELEASES.md
- Replaces `az_quickstart_9209()` with `az_quickstart_920101()` to prevent potential upgrade issues that could occur if future update hooks are needed in 2.0.x and establishes a new more granular update hook numbering scheme.
- Moves existing update hooks from `az_quickstart.profile` to `az_quickstart.install` to prevent performance impact on Drupal requests not related to install or update tasks.

## Related Issue
Closes #1350 

## How Has This Been Tested?
Locally with lando.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I've added this Pull Request to the AZ Quickstart Github project and set it to "Review in progress".
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

trackleft reviewed (dismissed):
Approved for 2.1.x

trackleft commented on az_quickstart.install:
```diff
@@ -49,3 +49,117 @@ function az_quickstart_install() {
+  // Update removed and moved to az_quickstart_update_92101().
+  // See https://github.com/az-digital/az_quickstart/issues/1350.
+}
+
```
I tested by adding a az_quickstart_update_9210() update on the 2.0.x branch

joeparsons commented on az_quickstart.install:
```diff
@@ -49,3 +49,117 @@ function az_quickstart_install() {
+/**
+ * Set the default columns classes on the new theme setting.
+ */
+function az_quickstart_update_92101() {
```
```suggestion
function az_quickstart_update_920101() {
```

joeparsons commented on az_quickstart.install:
```diff
@@ -49,3 +49,117 @@ function az_quickstart_install() {
+ * Set the default columns classes on the new theme setting (Removed).
+ */
+function az_quickstart_update_9209() {
+  // Update removed and moved to az_quickstart_update_92101().
```
```suggestion
  // Update removed and moved to az_quickstart_update_920101().
```

michaelhagedon commented on RELEASES.md:
```diff
@@ -53,20 +53,38 @@ The following types of changes are allowed for minor releases in addition to tho
+(e.g. `2.1.x`) as needed without changing the schema version sites to be
+something that would prevent this necessary  updates that are specific to a
+later minor release branch (e.g. `2.2.x`) from being applied when the site is
+upgraded to the later minor release.
```
I think there are some words missing from this paragraph, not sure exactly. 🤔

joeparsons replied:
Updated.  Thanks!

joeparsons commented on RELEASES.md:
```diff
@@ -53,20 +53,38 @@ The following types of changes are allowed for minor releases in addition to tho
+- ZZ = 2 digits for Quickstart minor version compatibility (e.g. `01`)
+- nn = 2 digits for sequential counting, starting with 01 (e.g. `01`)
+
+This will allow us to continue to add DB update hooks to a minor release branch
+(e.g. `2.1.x`) as needed without changing the schema version sites to be
+something that would prevent this necessary  updates that are specific to a
+later minor release branch (e.g. `2.2.x`) from being applied when the site is
+upgraded to the later minor release.
```
Hopefully this sounds better.

```suggestion
This will allow us to continue to add DB update hooks to a minor release branch
(e.g. `2.1.x`) as needed without updating a site's schema version to a number
that would prevent updates included in a later release branch to be applied when
the site is upgraded.
```

joeparsons commented:
## 2022-03-02 Updates

- Added update hook numbering convention guidelines to RELEASES.md
- Adjusted update hook number to follow updated numbering convention (changed `92101` to `920101`).

**This should be ready for final review/approval**.

Reminder:  This PR should be merged _before_ #1332.
